### PR TITLE
Add `Data.List.Relation.Binary.Sublist.Setoid` categorical properties

### LINKED ISSUE #816: Add category laws for sublists
` Data.List.Relation.Binary.Sublist.Setoid` defines the preorder of order-preserving embeddings of lists,
```agda
⊆-isPreorder : IsPreorder _≋_ _⊆_
⊆-isPreorder = record
  { isEquivalence = ≋-isEquivalence
  ; reflexive     = ⊆-reflexive
  ; trans         = ⊆-trans
  }
```
but it does not prove the category laws, which are very useful if one wants to use sublist-proofs as renamings in lambda calculus.
These laws would be:
- left unit: `⊆-trans ⊆-refl = id`
- right unit
- associativity of `⊆-trans`

Further, it would be useful to have the functor laws for `Data.List.Relation.Binary.Sublist.Propositional.lookup` in Data.List.Relation.Binary.Sublist.Propositional.Properties:
- identity: `lookup ⊆-refl = id`
- composition: `lookup (⊆-trans p q) = lookup q . lookup p`

### PR DESCRIPTION
**Pull request by jamesmckinna**
Fixes #816 

Includes some tidying up refactoring wrt `variable`s (leading to a lot of redundant `module _ where` declarations...) and `import`s.

UPDATED: rectification for consistency with existing proofs in `Data.List.Relation.Binary.Sublist.Propositional`:
* names of lemmas? DONE
* orientation of `⊆-assoc`? DONE

Possible TODO:
* pull proofs back to `Heterogeneous`? (not done; downstream PR?)

### CODE REVIEW

File: src/Data/List/Relation/Binary/Sublist/Setoid/Properties.agda [hunk 1]
```diff
@@ -279,6 +285,41 @@ module _ {x xs} where
...
+------------------------------------------------------------------------
+-- Categorical properties
+------------------------------------------------------------------------
+
+module _ (trans-reflˡ : ∀ {x y} (p : x ≈ y) → trans ≈-refl p ≡ p) where
+
+  left-unit : (pxs : xs ⊆ ys) → ⊆-trans ⊆-refl pxs ≡ pxs
+  left-unit [] = refl
+  left-unit (y ∷ʳ pxs) = cong (y ∷ʳ_) (left-unit pxs)
+  left-unit (x ∷ pxs) = cong₂ _∷_ (trans-reflˡ x) (left-unit pxs)
```
Comment (andreasabel) [inline src/Data/List/Relation/Binary/Sublist/Setoid/Properties.agda:297]: There used to be a `rewrite` which one can save by using `cong2`.

File: src/Data/List/Relation/Binary/Sublist/Setoid/Properties.agda [hunk 2]
```diff
@@ -279,6 +285,41 @@ module _ {x xs} where
...
+    []⊆-trans ps = []⊆-irrelevant (⊆-trans [] ps) ps
+
+  ⊆-assoc : (ps : as ⊆ bs) (qs : bs ⊆ cs) (rs : cs ⊆ ds) →
+            ⊆-trans (⊆-trans ps qs) rs ≡ ⊆-trans ps (⊆-trans qs rs)
+  ⊆-assoc [] [] rs rewrite []⊆-trans rs = ≡.sym ([]⊆-trans rs)
+  ⊆-assoc ps qs (_ ∷ʳ rs) = cong (_ ∷ʳ_) (⊆-assoc ps qs rs)
+  ⊆-assoc ps (_ ∷ʳ qs) (_ ∷ rs) = cong (_ ∷ʳ_) (⊆-assoc ps qs rs)
+  ⊆-assoc (_ ∷ʳ ps) (_ ∷ qs) (_ ∷ rs) = cong (_ ∷ʳ_) (⊆-assoc ps qs rs)
+  ⊆-assoc (p ∷ ps) (q ∷ qs) (r ∷ rs)
+    rewrite ≈-assoc p q r = cong ((trans p (trans q r)) ∷_ ) (⊆-assoc ps qs rs)
```
Comment (andreasabel) [inline src/Data/List/Relation/Binary/Sublist/Setoid/Properties.agda]: Make the split exact.

**Review by andreasabel — COMMENTED** [review]
I simplified the proofs a bit, getting rid of rewrite.

**Review by andreasabel — APPROVED** [review]
I added 4 micro commits which could be squashed.

**Comment by jamesmckinna** [comment]
> I added 4 micro commits which could be squashed.

Many thanks for the tweaks! FTR, `stdlib` commits get squashed in any case, so no worries about the size... ;-)

**Comment by andreasabel** [comment]
Greetings from AIM XXXVIII in Swansea!

**Comment by jamesmckinna** [comment]
Most recent commits now rectify the `Propositional` proofs as instances of the `Setoid` ones... but the latter have *explicit* parametrisation on the proofs of `xs⊆ ys` etc., while the former are *implicit*... 

... changing the latter would be `breaking`; but the former seems better suited to the proofs. Which one is 'right'?

**Review by JacquesCarette — APPROVED** [review]
Nice.

While it is entirely possible that there are more improvements that could be done in these files, these stand on their own as useful, so they can be done in a subsequent PR.

**Comment by JacquesCarette** [comment]
Tried to merge, but the build failed *at the HTML stage*. So it's probably just bad luck, and it should be tried again.

**Pull request merged**